harness/determined/cli/workspace.py

```diff
@@ -267,9 +267,9 @@ def create_workspace(args: argparse.Namespace) -> None:
     agent_user_group = _parse_agent_user_group_args(args)
     checkpoint_storage = _parse_checkpoint_storage_args(args)
 
-    if args.cluster_name and not (
-        args.namespace or args.auto_create_namespace or args.auto_create_namespace_all_clusters
-    ):
+    auto_create_namespace = args.auto_create_namespace or args.auto_create_namespace_all_clusters
+    set_namespace = args.namespace or auto_create_namespace
+    if args.cluster_name and not set_namespace:
```

Comment:
I love this change; I think it's really easy to follow!

Reply:
Nice work. It's easier to understand the intention!